packages/hash/frontend/src/blocks/page/Comments/CommentTextField.tsx

```diff
@@ -22,44 +16,87 @@ import {
   FontAwesomeIcon,
   LoadingSpinner,
 } from "@hashintel/hash-design-system";
-import { faComment } from "@fortawesome/free-regular-svg-icons";
 import { TextToken } from "@hashintel/hash-shared/graphql/types";
-import { TEXT_TOKEN_PROPERTY_TYPE_BASE_URI } from "@hashintel/hash-shared/entityStore";
-
-import { textBlockNodeToEntityProperties } from "@hashintel/hash-shared/text";
+import {
+  textBlockNodeToTextTokens,
+  textBlockNodesFromTokens,
+} from "@hashintel/hash-shared/text";
+import { isEqual } from "lodash";
 import { usePortals } from "../usePortals";
 import { createFormatPlugins } from "../createFormatPlugins";
 import {
   createSuggester,
   suggesterPluginKey,
 } from "../createSuggester/createSuggester";
 import { useRouteAccountInfo } from "../../../shared/routing";
-import styles from "../style.module.css";
-import { commentPlaceholderPlugin } from "./commentPlaceholderPlugin";
+import styles from "./style.module.css";
+import {
+  CommentPlaceholderAction,
+  commentPlaceholderPlugin,
+  commentPlaceholderPluginkey,
+} from "./commentPlaceholderPlugin";
 import { createTextEditorView } from "../createTextEditorView";
 
 type CommentTextFieldProps = {
-  onClose: () => void;
-  onSubmit: (content: TextToken[]) => Promise<void>;
+  value?: TextToken[];
+  placeholder?: string;
+  className?: string;
+  loading?: boolean;
+  editable?: boolean;
+  readOnly?: boolean;
+  onChange?: (value: TextToken[]) => void;
+  onLineCountChange?: (lines: number) => void;
+  onFocusChange?: (focused: boolean) => void;
+  onClose?: () => void;
+  onSubmit?: () => Promise<void>;
 };
 
+const LINE_HEIGHT = 21;
+
 export const CommentTextField: FunctionComponent<CommentTextFieldProps> = ({
+  value,
+  placeholder = "Leave a comment",
+  className = "",
+  loading = false,
+  editable = false,
+  readOnly = false,
+  onChange,
+  onLineCountChange,
+  onFocusChange,
   onClose,
   onSubmit,
 }) => {
   const viewRef = useRef<EditorView<Schema>>();
   const [portals, renderPortal] = usePortals();
   const { accountId } = useRouteAccountInfo();
   const editorContainerRef = useRef<HTMLDivElement>();
-  const loadingRef = useRef(false);
-  const [loading, setLoading] = useState(false);
+  const editableRef = useRef(false);
   const eventsRef = useRef({ onClose, onSubmit });
 
   useLayoutEffect(() => {
     eventsRef.current = { onClose, onSubmit };
-    loadingRef.current = loading;
+    editableRef.current = editable;
+
+    if (viewRef.current && onLineCountChange) {
+      onLineCountChange(
```

Comment:
This ought to be setup with a resize observer, rather than relying on an effect happening to trigger when a resize occurs. I don't think this blocks this PR, but leaving this thought here.